### PR TITLE
Feature/mx 1660 ldap search endpoint

### PR DESCRIPTION
**Pull request by vyvytranngoc**
# PR Context

-  LDAP search endpoint in mex-backend
- mex-backend has a new search endpoint returning ldap persons
# Added

- extract ldap: get_persons methods and tests

# Changes

# Deprecated

# Removed

# Fixed

# Security

### CODE REVIEW

File: tests/ldap/test_extract.py [hunk 1]
```diff
@@ -173,3 +177,51 @@ def test_get_merged_ids_by_query_string(
         ldap_persons_with_query, ldap_primary_source
     )
     assert merged_ids_by_query_string == expected
+
+
+def test_get_persons_mocked(ldap_mocker: LDAPMocker) -> None:
```
Comment (cutoffthetop) [inline tests/ldap/test_extract.py:182]:
```suggestion
def test_get_persons_by_name_mocked(ldap_mocker: LDAPMocker) -> None:
```

File: tests/ldap/test_extract.py [hunk 2]
```diff
@@ -173,3 +177,51 @@ def test_get_merged_ids_by_query_string(
...
+        "objectGUID": UUID("00000000-0000-4000-8000-000000000000"),
+        "ou": ["XY"],
+        "sAMAccountName": "SampleS",
+        "sn": "Sample",
+    }
+    assert len(persons_list) == 1
+    assert persons_list[0].model_dump(exclude_none=True)["givenName"] == expected["givenName"]
+
+
+def test_get_count_persons_mocked(ldap_mocker: LDAPMocker) -> None:
```
Comment (cutoffthetop) [inline tests/ldap/test_extract.py:203]:
```suggestion
def test_get_count_of_found_persons_by_name_mocked(ldap_mocker: LDAPMocker) -> None:
```

File: tests/ldap/test_extract.py [hunk 3]
```diff
@@ -173,3 +177,51 @@ def test_get_merged_ids_by_query_string(
...
+    ldap_mocker([[SAMPLE_PERSON_ATTRS]])
+    persons_count = get_count_of_found_persons_by_name(
+        surname="Sample", given_name="Sam"
+    )
+    assert persons_count == 1
+
+
+def test_get_person_mocked(ldap_mocker: LDAPMocker) -> None:
```
Comment (cutoffthetop) [inline tests/ldap/test_extract.py:211]:
```suggestion
def test_get_person_by_id_mocked(ldap_mocker: LDAPMocker) -> None:
```

best if the original name of the function you are testing is reflected in the test, so it's easier to find and to indentify

File: mex/common/ldap/extract.py [hunk 4]
```diff
@@ -111,3 +112,61 @@ def get_merged_ids_by_query_string(
                 MergedPersonIdentifier(identities[0].stableTargetId)
             )
     return merged_ids_by_attribute
+
+
+def get_person_by_id(
+    objectGUID: str = "*",
+    employeeID: str = "*",
```
Comment (cutoffthetop) [inline mex/common/ldap/extract.py:119]:
```suggestion
    objectGUID: str = "*",  # noqa: N803
    employeeID: str = "*",  # noqa: N803
```

File: mex/common/ldap/extract.py [hunk 5]
```diff
@@ -111,3 +112,61 @@ def get_merged_ids_by_query_string(
...
+    """Search for a unique person in LDAP.
+
+    Args:
+        objectGUID: Internal LDAP identifier.
+        employeeID: Employee ID, must be present.
+        **filters: Filters for LDAP search.
+
+    Returns:
+        Single LDAP person matching the filters.
+
```
Comment (cutoffthetop) [inline mex/common/ldap/extract.py:131]:
```suggestion
```

**Review by cutoffthetop — CHANGES_REQUESTED** [review]
looks good, just a few clean-up suggestions

**Comment by vyvytranngoc** [comment]
ruff-linter fails. Could not comprehend which files were re-formatted. Therefore, I re-run ruff linter in .\tests\ldap and .\mex\common\ldap. But there all checks passed.

**Comment by cutoffthetop** [comment]
> ruff-linter fails. Could not comprehend which files were re-formatted. Therefore, I re-run ruff linter in .\tests\ldap and .\mex\common\ldap. But there all checks passed.

@vyvytranngoc you can run `pdm lint` to run all pre-commit checks and linters on all files -- this should give you the same errors / re-formatting as the pipeline

**Review by cutoffthetop — APPROVED** [review]

**Pull request merged**